### PR TITLE
[JENKINS-43891] Use 2.X parent POM

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-43891](https://issues.jenkins-ci.org/browse/JENKINS-43891)

- Use 2.X parent POM.
- Bump core version.
- Bump dependencies versions.

@reviewbybees @stephenc @jglick @kwhetstone @andresrc

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Review by kwhetstone — APPROVED** [review]
:bee:

File: pom.xml
Comment (jglick) [inline pom.xml]: Delete, this is the default.

**Review by jglick — APPROVED** [review]

File: pom.xml
Comment (varyvol) [reply pom.xml]: Right, removed.

**Review by jglick — APPROVED** [review]

**Review by andresrc — APPROVED** [review]

**Comment by varyvol** [comment]
@christ66 @stephenc could you review and merge this if appropriate?

**Comment by varyvol** [comment]
@stephenc @christ66 would it be possible to release a new version including these changes?

**Comment by christ66** [comment]
@varyvol Add yourself to the list of maintainers: https://github.com/jenkins-infra/repository-permissions-updater/blob/50518739b172275148527eab496ccab6bc5befdc/permissions/plugin-deployed-on-column.yml. Currently there are no maintainers for this plugin.

**Pull request merged**
